File: src/main/java/jb/trainer/SvmTrainer.java
package jb.trainer;

import de.bwaldvogel.liblinear.Linear;
import de.bwaldvogel.liblinear.Model;
import de.bwaldvogel.liblinear.Parameter;
import jb.config.Opts;
import jb.data.Dataset;

import java.util.ArrayList;
import java.util.List;

public class SvmTrainer implements Trainer{

    @Override
    public List<Model> train(Dataset dataset, Opts opts) {
        Parameter parameter = new Parameter(opts.getSolverType(), opts.getC(), opts.getEps());
        List<Model> clfs = new ArrayList<>();
        for (jb.data.Problem problem : dataset.getProblems()) {
            clfs.add(Linear.train(problem.convertToLibLinearProblem(opts), parameter));
        }
        return clfs;
    }

}

File: src/main/java/jb/trainer/SvmTrainer.kt
package jb.trainer

import de.bwaldvogel.liblinear.Linear
import de.bwaldvogel.liblinear.Model
import de.bwaldvogel.liblinear.Parameter
import jb.config.Opts
import jb.data.Dataset

import java.util.ArrayList

class SvmTrainer : Trainer {

    override fun train(dataset: Dataset, opts: Opts): List<Model> {
        val parameter = Parameter(opts.solverType, opts.c, opts.eps)
        val clfs = ArrayList<Model>()
        for (problem in dataset.problems) {
            clfs.add(Linear.train(problem.convertToLibLinearProblem(opts), parameter))
        }
        return clfs
    }

}
